reframe/utility/__init__.py

```diff
@@ -594,6 +594,8 @@ def find_modules(substr, environ_mapping=None):
     This function is a generator and will yield tuples of partition,
     environment and module combinations for each partition of the current
     system and for each environment of a partition.
+    In case of multiple module entries sharing the same name, only the first
+    one is considered.
```

Comment:
I don't think that this sentence offers anything to the user. We are fixing a bug, since this was the expected behaviour.